UnityProject/Assets/Scripts/Player/PlayerScript.cs

```diff
@@ -757,15 +765,107 @@ public void SyncPossessingID(uint previouslyPossessing, uint currentlyPossessing
 		Itself.PreImplementedSyncPossessingID(previouslyPossessing, currentlyPossessing);
 	}
 
+	#region FAITH
+
+	[Server]
+	public void JoinReligion(Faith newFaith)
+	{
+		LeaveReligion();
+		currentFaith = newFaith;
+		FaithName = currentFaith.FaithName;
+		foreach (var prop in currentFaith.FaithProperties)
+		{
+			prop.OnJoinFaith(this);
+		}
+		if (FaithManager.Instance.CurrentFaith == newFaith) FaithManager.Instance.FaithMembers.Add(this);
```

Comment:
how does equality on faiths work? Are faiths ``record``? Do you implement any equality interface?

Reply:
it uses unity object references, so it does its own stuff; we don't really need to implement our own checks unless we're creating these from scratch; but since they're initially tied to SOs it shouldn't matter.

Reply:
did you test it? Because by default this would check if both faiths are the same instanced object, not really if they conceptually are the same faith

Reply:
It seems to work when I checked it using MPM? I'll check again after playtest.

Reply:
tested this btw, it works